src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -797,348 +186,163 @@ const std::vector<impl_desc_type>& FullyConnected::getDefaultImplPriority() {
     }
-    // 3D FC requires implicit reshape so strides should be defined
-    auto supportsUndefStridesAndOffset = [&]() {
-        return getOutputShapeAtPort(0).getRank() == 2;
-    };
-
-    auto addSupportedPrimitiveDescriptor = [&](const dnnl::primitive_desc& prim_desc) {
-        std::vector<PortConfig> inConfs, outConfs;
-        const int inPlaceOutPort = canBeInPlace() ? 0 : -1;
-
-        for (size_t i = 0; i < descInputNumbers(); i++) {
-            auto desc = getSrcMemDesc(prim_desc, i);
-            const auto inputBlockedMask = (supportsUndefStridesAndOffset() && !(i == WEIGHTS_ID && useSparseWeights)) ?
-                BlockedMemoryDesc::EMPTY_MASK :
-                BlockedMemoryDesc::FULL_MASK;
-
-            inConfs.emplace_back(desc, inputBlockedMask);
-        }
 
-        const auto outputBlockedMask = supportsUndefStridesAndOffset() ? BlockedMemoryDesc::EMPTY_MASK : BlockedMemoryDesc::FULL_MASK;
+    DEBUG_LOG("elementsCount = ",
+              elementsCount,
+              ", zerosCount = ",
+              zerosCount,
+              ", nnzCount = ",
+              elementsCount - zerosCount);
 
-        for (size_t i = 0; i < descOutputNumbers(); i++) {
-            auto desc = getDstMemDesc(prim_desc, i);
+    auto sparseRate = static_cast<float>(zerosCount) / static_cast<float>(elementsCount);
 
-            outConfs.emplace_back(desc, outputBlockedMask, inPlaceOutPort);
-        }
+    DEBUG_LOG("Sparse rate = ",
+              sparseRate * 100,
+              "%, min sparse rate = ",
+              minSparseRate * 100,
+              "%, use sparse weights = ",
+              sparseRate >= minSparseRate);
 
-        const NodeConfig config(inConfs, outConfs);
-        const impl_desc_type impl_type = parse_impl_name(prim_desc.impl_info_str());
+    return sparseRate < minSparseRate;
+}
 
-        supportedPrimitiveDescriptors.emplace_back(config, impl_type);
+void FullyConnected::initSupportedPrimitiveDescriptors() {
+    attrs.withBias = getOriginalInputsNumber() == 3;
+    attrs.dequantizationScales = getDQScales();
+    attrs.sparseWeights = useSparseWeightsDecompression(getParentEdgeAt(DATA_ID)->getParent(),
+                                                        getOriginalInputPrecisionAtPort(DATA_ID),
+                                                        context->getConfig().fcSparseWeiDecompressionRate);
+    postOps = getPostOps(fusedWith);
+
+    const auto& srcTypes = getOriginalInputPrecisions();
+    auto dstTypes = getOriginalOutputPrecisions();
+    // @todo graph optimizer should update original output precisions instead
+    if (!fusedWith.empty())
+        dstTypes = fusedWith.back()->getOriginalOutputPrecisions();
+
+    MemoryDescs srcDescs;
+    const auto& creatorsMap = BlockedDescCreator::getCommonCreators();
+    for (size_t i = 0; i < srcTypes.size(); i++) {
+        const auto srcDesc = creatorsMap.at(LayoutType::ncsp)->createSharedDesc(srcTypes[i], getInputShapeAtPort(i));
```

Comment:
we do not really care about layout type here, just creating memory descriptor with port precision

Reply:
Any layout would help here

Reply:
Just wondering how multy-layout cases are expected to be handled. Like for Eltwise node we should support planar, per-channel and blocked layouts. My expectation is that all 3 Node configurations should be placed into supportedPrimitiveDescriptors to help Graph choose best instance based on parents layout.
Or expections is that `getProperMemoryDescriptors` also returns descs with ANY layout and fallback logic will insert Reorders if necessary?

Reply:
For multilayout nodes we will have to extend the layout mapping the same way it is done for type mapping.
Also, in order to force some layout (i.e. perchannel) the corresponding implementation should be added to the list of the implementations. So, basically, in case of the `eltwise` there will be two almost identical implementations in the list, the only difference is one will be forcing perchannel layout as part of its `requiresFallback` logic.